spec/recipients_spec.rb

```diff
@@ -29,21 +29,33 @@
         test_email = Mailer.test_email to: "to1@example.com, to2@example.com"
         @delivery_method.deliver!(test_email)
 
-        expect(@delivery_method.data[:recipients]).to eq([{address: {email: "to1@example.com"}}, {address: {email: "to2@example.com"}}])
+        expect(@delivery_method.data[:recipients]).to match([a_hash_including({address: {email: "to1@example.com", header_to: anything}}),
+                                                             a_hash_including({address: {email: "to2@example.com", header_to: anything}})])
       end
 
       it "handles name and email" do
         test_email = Mailer.test_email to: "Sam Test <to1@example.com>, Joe Test <to2@example.com>"
         @delivery_method.deliver!(test_email)
 
-        expect(@delivery_method.data[:recipients]).to eq([{:address=>{:email=>"to1@example.com", :name=>"Sam Test"}}, {:address=>{:email=>"to2@example.com", :name=>"Joe Test"}}])
+        expect(@delivery_method.data[:recipients]).to match([a_hash_including({:address=>{:email=>"to1@example.com", :name=>"Sam Test", header_to: anything}}),
+                                                             a_hash_including({:address=>{:email=>"to2@example.com", :name=>"Joe Test", header_to: anything}})])
       end
 
       it "handles mix of email only and name/email" do
         test_email = Mailer.test_email to: "Sam Test <to1@example.com>, to2@example.com"
         @delivery_method.deliver!(test_email)
 
-        expect(@delivery_method.data[:recipients]).to eq([{:address=>{:email=>"to1@example.com", :name=>"Sam Test"}}, {:address=>{:email=>"to2@example.com"}}])
+        expect(@delivery_method.data[:recipients]).to match_array([a_hash_including({:address=>{:email=>"to1@example.com", :name=>"Sam Test", header_to: anything}}),
+                                                                   a_hash_including({:address=>{:email=>"to2@example.com", header_to: anything}})])
+      end
+
+      it "compiles list of email addresses to populate :header_to for each recipient" do
+        expected_header_to = "a@a.com,b@b.com"
+        test_email = Mailer.test_email to: "a <a@a.com>, b@b.com"
+        @delivery_method.deliver!(test_email)
+
+        expect(@delivery_method.data[:recipients].first[:address][:header_to]).to eql(expected_header_to)
+        expect(@delivery_method.data[:recipients].second[:address][:header_to]).to eql(expected_header_to)
```

Comment:
Test that header_to is set properly on all to-recipients.

The other tests above (and in the other spec files) were modified to match "a_hash_including", so that way we could test for "anything" for all to-recipient header_to without relying on a specific value.